Formula/csound.rb

```diff
@@ -99,43 +112,68 @@ def install
-                      "-DBUILD_MP3OUT_OPCODE=ON",
-                      "-DBUILD_OPENCL_OPCODES=OFF",
-                      "-DBUILD_P5GLOVE_OPCODES=ON",
-                      "-DBUILD_PYTHON_OPCODES=ON",
-                      "-DBUILD_STK_OPCODES=ON",
-                      "-DBUILD_WEBSOCKET_OPCODE=ON",
-                      "-DBUILD_WIIMOTE_OPCODES=ON",
-                      "-DCSOUND_FRAMEWORK=#{frameworks}/CsoundLib64.framework",
-                      "-DCSOUND_INCLUDE_DIR=#{include}/csound",
-                      "-DGMM_INCLUDE_DIR=#{buildpath}",
-                      "-DPLUGIN_INSTALL_DIR=#{frameworks}/CsoundLib64.framework/Resources/Opcodes64",
-                      "-DUSE_FLTK=ON",
-                      *std_cmake_args
+      args = [
+        "-DABLETON_LINK_HOME=#{buildpath}/ableton-link",
+        "-DBUILD_ABLETON_LINK_OPCODES=ON",
+        "-DBUILD_CHUA_OPCODES=ON",
+        "-DBUILD_CUDA_OPCODES=OFF",
+        "-DBUILD_FAUST_OPCODES=ON",
+        "-DBUILD_FLUID_OPCODES=ON",
+        "-DBUILD_HDF5_OPCODES=ON",
+        "-DBUILD_IMAGE_OPCODES=ON",
+        "-DBUILD_JACK_OPCODES=ON",
+        "-DBUILD_LINEAR_ALGEBRA_OPCODES=ON",
+        "-DBUILD_MP3OUT_OPCODE=ON",
+        "-DBUILD_OPENCL_OPCODES=OFF",
+        "-DBUILD_PYTHON_OPCODES=ON",
+        "-DBUILD_STK_OPCODES=ON",
+        "-DBUILD_WEBSOCKET_OPCODE=ON",
+        "-DGMM_INCLUDE_DIR=#{buildpath}",
+        "-DUSE_FLTK=ON",
+      ]
+      if OS.mac?
+        args << "-DBUILD_P5GLOVE_OPCODES=ON"
+        args << "-DBUILD_WIIMOTE_OPCODES=ON"
+        args << "-DCSOUND_FRAMEWORK=#{frameworks}/CsoundLib64.framework"
+        args << "-DCSOUND_INCLUDE_DIR=#{include}/csound"
+        args << "-DPLUGIN_INSTALL_DIR=#{frameworks}/CsoundLib64.framework/Resources/Opcodes64"
+      else
+        args << "-DBUILD_P5GLOVE_OPCODES=OFF"
+        args << "-DBUILD_WIIMOTE_OPCODES=OFF"
+      end
+      system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
       system "cmake", "--build", "build"
       system "cmake", "--install", "build"
     end
   end
 
   def caveats
-    <<~EOS
+    caveats = <<~EOS
       To use the Java bindings, you may need to add to #{shell_profile}:
         export CLASSPATH="#{opt_libexec}/csnd6.jar:."
       and link the native shared library into your Java Extensions folder:
-        mkdir -p ~/Library/Java/Extensions
-        ln -s "#{opt_libexec}/lib_jcsound6.jnilib" ~/Library/Java/Extensions
     EOS
+
+    on_macos do
+      caveats = <<~EOS
+        #{caveats}\
+          mkdir -p ~/Library/Java/Extensions
+          ln -s "#{opt_libexec}/lib_jcsound6.jnilib" ~/Library/Java/Extensions
+      EOS
+    end
+
+    on_linux do
+      caveats = <<~EOS
+        srconv is not installed because it conflicts with binutils. To run srconv:
+          csound --utility=srconv
+
+        #{caveats}\
+          sudo mkdir -p /usr/java/packages/lib
+          sudo ln -s "#{opt_libexec}/lib_jcsound6.jnilib" /usr/java/packages/lib
+      EOS
+    end
+
```

Comment:
Sorry to nitpick here, but is there a directory in `HOMEBREW_PREFIX` that users can link the java bindings to that will be picked up as a Java extension?  Or does our OpenJDK hardcode that directory as the only place to find extensions?

Reply:
> …is there a directory in HOMEBREW_PREFIX that users can link the java bindings to that will be picked up as a Java extension?

Not that I know of.

> …does our OpenJDK hardcode that directory as the only place to find extensions?

As far as I know, Java looks for extensions in the directories defined by its `java.library.path` setting. When I run

```sh
$(brew --prefix openjdk)/bin/java -XshowSettings:properties
```

in a Terminal on Ubuntu 22.04, I see that the value of `java.library.path` is

```
/usr/java/packages/lib
/usr/lib64
/lib64
/lib
/usr/lib
```

The first path seemed like the best choice.

This caveat is based on the Subversion formula (200,000+ installs per year):

https://github.com/Homebrew/homebrew-core/blob/afb158ad34ebd8966dd25ce21bc3a130f485bcef/Formula/subversion.rb#L145-L147

Reply:
It looks like on Linux, by default `java.library.path` is `/usr/java/packages/lib:$LD_LIBRARY_PATH`, so the user could specify the path to the `libexec` folder with `LD_LIBRARY_PATH`, although we don't like to encourage the use of the variable as it can break things.  It looks like it may be possible to pass `-Djava.library.path` when calling the JVM as well.

@Homebrew/core what do we think about this caveat on Linux? Should we at some point try to tweak OpenJDK so it can search some place like `HOMEBREW_PREFIX/lib/java/packages` that any formula can link into.  I don't want to hold up this PR any longer than necessary but I think this is worth getting feedback on.

Reply:
> …the user could specify the path to the `libexec` folder with `LD_LIBRARY_PATH`, although we don't like to encourage the use of the variable as it can break things.

I figure I should mention that using (what I believe to be) a similar environment variable on macOS (`DYLD_FRAMEWORK_PATH`) was removed from this very formula [relatively recently](https://github.com/Homebrew/homebrew-core/pull/89990).

> …it may be possible to pass `-Djava.library.path` when calling the JVM…

It is; that’s what’s happening in the tests:

https://github.com/Homebrew/homebrew-core/blob/eafc0d300778825bd2b3e7003af91ec136fcf4fa/Formula/csound.rb#L247-L248

Reply:
I'm sufficiently convinced that this is not the result of some underlying issue with our OpenJDK setup and since there has been no other feedback I'm going to merge this and get `csound` bottled.  This caveat can be changed in a syntax-only PR later if needed.